src/meta/src/stream/stream_manager.rs

```diff
@@ -504,8 +511,13 @@ impl GlobalStreamManager {
             existing_locations,
         }: ReplaceTableContext,
     ) -> MetaResult<()> {
-        self.build_actors(&table_fragments, &building_locations, &existing_locations)
-            .await?;
+        self.build_actors(
```

Comment:
IIUC, for `replace_table`, the table_id of the new TableFragment should be the same as the `old_table_fragments`. You can add an `assert_eq` to verify this. If this is true, in `build_actors`, we can always use the `table_id` of `table_fragments` and we don't need to pass the separate table_id to the method.